cupy/cuda/function.pyx

```diff
@@ -64,8 +67,10 @@ cdef inline CPointer _pointer(x):
     cdef Py_ssize_t itemsize
     if x is None:
         return CPointer()
-    if hasattr(x, 'cstruct'):
```

Comment:
Currently only core.ndarray and core.Indexer have cstruct attribute (in my understanding), so within the scope of cupy, it doesn't break backward compatibility. It would be a condition that never be satisfied while costly. Is the statement still needed?